Realm/RLMRealm.mm

```diff
@@ -780,4 +787,158 @@ - (void)detachAllEnumerators {
     _collectionEnumerators = nil;
 }
 
+#pragma mark - Async Writes
+
+- (void)writeAsyncWithBlock:(RLMAsyncWriteBlock)block completion:(RLMCompletionBlock)completion {
+    RLMRealmConfiguration *config = self.configuration;
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        NSError *error = nil;
+        RLMRealm *bgRealm = [RLMRealm realmWithConfiguration:config error:&error];
+        
+        if (error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                completion(error);
+            });
+            return;
+        }
+        
+        [bgRealm transactionWithBlock:^{
+            block(bgRealm);
+        } error:&error];
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            // Should refresh be called before completion?
+            completion(error);
+        });
+    });
+}
+
+- (void)writeObjectAsync:(RLMObject *)object withBlock:(RLMAsyncWriteObjectBlock)block completion:(RLMCompletionBlock)completion {
+    if (object.realm->_realm != self->_realm) {
+        @throw RLMException(@"Object is not persisted in this Realm");
+    }
+    
+    auto& sg = object.realm->_realm->shared_group();
+    __block std::unique_ptr<realm::SharedGroup::Handover<realm::Row> > handover_row = sg.export_for_handover(object->_row);
+    
+    RLMRealmConfiguration *config = object.realm.configuration;
+    Class accessorClass = object.objectSchema.accessorClass;
+    RLMObjectSchema *objectSchema = object.objectSchema;
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        NSError *error = nil;
+        RLMRealm *newRealm = [RLMRealm realmWithConfiguration:config error:&error];
+        
+        if (error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                completion(error);
+            });
+            return;
+        }
+        
+        newRealm->_realm->read_group();
+        auto& new_sg = newRealm->_realm->shared_group();
+        std::unique_ptr<realm::Row> new_row( new_sg.import_from_handover(move(handover_row)) );
```

Comment:
This will fail for any nontrivial cases of writes being made on multiple threads (e.g. if asyncWrite is called twice in a row...) because objects can only be handed over to a sharedgroup at the same version as the source one.